File: SignatureRequests/SignatureRequests.Webs/Scripts/react/src/Entities/SignatureEntity.ts
import UserEntity from "./UserEntity";

class SignatureEntity {
  id: number;
  imagePath: string;
  certificatePath: string;
  certificatePassword: string;
  location?: string;
  reason?: string;
  contactInfo?: string;
  isInitial: boolean;
  user: UserEntity;
  expirationDate?: Date;

  constructor(data: any) {
    this.id = data.Id;
    this.imagePath = data.ImagePath;
    this.certificatePath = data.CertificatePath;
    this.certificatePassword = data.CertificatePassword;
    this.location = data.Location;
    this.reason = data.Reason;
    this.contactInfo = data.ContactInfo;
    this.isInitial = data.isInitial;
    this.user = data.User;
    this.expirationDate = data.ExpirationDate;
  }
}

export default SignatureEntity;

File: SignatureRequests/SignatureRequests.Core/Entities/SignatureEntity.cs
﻿using BTSuggestions.Core.Entities;
using Newtonsoft.Json;
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace SignatureRequests.Core.Entities
{
    public class SignatureEntity: BaseEntity
    {
        [Required]
        public string ImagePath { get; set; }

        [Required] 
        public string CertificatePath { get; set; }

        [MaxLength(255), Required]
        public string CertificatePassword { get; set; }

        [MaxLength(255)]
        public string Location { get; set; }

        [MaxLength(400)]
        public string Reason { get; set; }

        [MaxLength(400)]
        public string ContactInfo { get; set; }

        [Required]
        public Boolean isInitial { get; set; }

        [Required]
        public virtual UserEntity User { get; set; }

        public int UserId { get; set; }

        public DateTime ExpirationDate { get; set; }

    }
}
